bolt/lib/Rewrite/BinaryPassManager.cpp

```diff
@@ -257,6 +257,56 @@ static cl::opt<bool> CMOVConversionFlag("cmov-conversion",
                                         cl::ReallyHidden,
                                         cl::cat(BoltOptCategory));
 
+static cl::opt<bool> DisableLongJmp("disable-longjmp-pass",
+                                    cl::desc("Disable the long jump pass"),
+                                    cl::ReallyHidden, cl::cat(BoltOptCategory));
+
+static cl::opt<bool> DisableBranchFix("disable-branch-fix-pass",
+                                      cl::desc("Disable the branch fix pass"),
+                                      cl::ReallyHidden,
+                                      cl::cat(BoltOptCategory));
+
+static cl::opt<bool> DisableAligner("disable-aligner-pass",
+                                    cl::desc("Disable the aligner pass"),
+                                    cl::ReallyHidden, cl::cat(BoltOptCategory));
+
+static cl::opt<bool> DisableVeneer("disable-veneer-pass",
+                                   cl::desc("Disable the veneer pass"),
+                                   cl::ReallyHidden, cl::cat(BoltOptCategory));
+
+static cl::opt<bool>
+    DisableUnreachable("disable-unreachable-pass",
+                       cl::desc("Disable the unreachable pass"),
+                       cl::ReallyHidden, cl::cat(BoltOptCategory));
+
+static cl::opt<bool>
+    DisableAnnotations("disable-annotations-pass",
+                       cl::desc("Disable the annotations pass"),
+                       cl::ReallyHidden, cl::cat(BoltOptCategory));
+
+static cl::opt<bool>
+    DisableNormalizeCFG("disable-normalize-cfg-pass",
+                        cl::desc("Disable the normalize CFG pass"),
+                        cl::ReallyHidden, cl::cat(BoltOptCategory));
+
+static cl::opt<bool>
+    DisableFunctionReorder("disable-function-reorder-pass",
+                           cl::desc("Disable the function reorder pass"),
+                           cl::ReallyHidden, cl::cat(BoltOptCategory));
+
+static cl::opt<bool>
+    DisableShortenInstruction("disable-shorten-instruction-pass",
```

Comment:
I guess this pass is not mandatory, but it is always enabled because it is usually always a net win. If you want to create a flag to disable it, I suggest creating a flag to enable it "ShortenInstructions" that is default-true, and pass that to the binary pass manager like other passes. Then you can pass --shorten-instructions=false if you want to disable it.